workflow/rules/trimming.smk

```diff
@@ -10,25 +12,48 @@ rule get_sra:
 
 def get_raw_fastq(wildcards):
     unit = units.loc[wildcards.sample].loc[wildcards.unit]
+    if unit["fq1"].endswith("gz"):
+        ending = ".gz"
+    else:
+        ending = ""
+
     if pd.isna(unit["fq1"]):
         # SRA sample (always paired-end for now)
         accession = unit["sra"]
         return expand("sra/{accession}_{read}.fastq", accession=accession, read=[1, 2])
     if pd.isna(unit["fq2"]):
         # single end local sample
-        return unit["fq1"].values
+        return f"pipe/cutadapt/{unit.sample_name}-{unit.unit_name}.fq1.fastq{ending}"
     else:
         # paired end local sample
-        return unit[["fq1", "fq2"]].values
+        return expand(f"pipe/cutadapt/{unit.sample_name}-{unit.unit_name}.{{read}}.fastq{ending}", read=["fq1","fq2"])
+
+
+def cutadapt_pipe_input(wc):
+    files = list(sorted(glob.glob(units.loc[wc.sample].loc[wc.unit, wc.fq])))
+    #print(wc, units.loc[wc.sample].loc[wc.unit, wc.fq])
+    assert(len(files) > 0)
+    return files
+
+
+rule cutadapt_pipe:
+    threads:
+        0
```

Comment:
Always put threads directly above the shell directive. It is most related to that, and less important for reading the step than input and output.